colossalai/auto_parallel/solver/sharding_strategy.py

```diff
@@ -37,20 +40,36 @@ class ShardingStrategy:
     input_shardings: List[ShardingSpec] = None
 
 
+class OperandType(Enum):
+    """
+    An operand can come from the argument list of an operator or the parameter list of a module.
+    """
+    ARG = 0
+    PARAM = 1
+
+
+@dataclass
+class Operand:
+    name: str
+    type: OperandType
+
+
 @dataclass
-class TrainCycleItem:
+class TrainCycleCost:
     """
     TrainCycleItem is a dataclass to store the items which have different values for the forward and backward pass
     in a training iteration.
 
     Args:
-        fwd (Any): the item for the forward pass
-        bwd (Any): the item for the backward pass
-        total (Any): the total value for the forward and backward pass
+        fwd (float): the item for the forward pass
+        bwd (float): the item for the backward pass
     """
-    fwd: Any
-    bwd: Any
-    total: Any
+    fwd: float
+    bwd: float
+
+    @property
+    def total(self):
+        return self.fwd + self.bwd
```

Comment:
And we can't samply add the fwd and bwd cost together, such as total memory cost is not the sum of bwd memory cost and fwd memory cost